src/metabase/native_query_analyzer/replacement.clj

```diff
@@ -53,17 +60,25 @@
 
       :else
       (let [v         (param->value (:k token))
-            card-ref? (params/ReferencedCardQuery? v)]
+            card-ref? (params/ReferencedCardQuery? v)
+            snippet?  (params/ReferencedQuerySnippet? v)]
         (cond
           card-ref?
           (let [sub (gen-table-sentinel raw-query)]
             (recur rest
                    (str query-so-far sub)
                    (add-tag substitutions sub token)))
 
+          snippet?
+          (let [sub (gen-snippet-sentinel raw-query v)]
+            (recur rest
+                   (str query-so-far sub)
+                   (add-tag substitutions sub token)))
+
           ;; Plain variable
           (and (params/Param? token)
-               (not card-ref?))
+               (not card-ref?)
+               (not snippet?))
```

Comment:
I don't want to upset the CI applecart, but next time I'm in this file (which will be any minute now) I can get rid of these `not`s and just put in a comment saying that the order of clauses matters.

Reply:
Snap 🫰